### DISCUSSION TITLE
Changed RPI, same SD, but lost pinout definition.

### DISCUSSION THREAD

**Discussion by davthomaspilot** [tagyoureit/nodejs-poolController#977] [REM]
I switched to a new RPI and booted using the same SD.  I've lost the pinout for the Atlas Scientific ph and ORP probes.  And the dosing pump relay connections.

Is there somewhere I can find what the were previously?

**Comment by davthomaspilot** [comment]
I now recall the Atlas Scientific ORP and PH probes were on the i2c bus.  I clicked the + sign and a new tabe showed up for I2C - Bus #1.

I then clicke the rescan and visited that tab and saw this:

![image](https://github.com/tagyoureit/nodejs-poolController/assets/6830886/88097eaa-923d-42e8-b3ca-a65bc33f0efc)

I don't recall adding the text that described the ORP and PH probes.  Is that read over the I2C bus (from the device), or is it "normal" for me to type something descriptive?

**Reply by davthomaspilot** [reply]
Disregard, I found the pull-down.  I guess I'll try assigning one to the ORP and the other address to PH.  If they work great, else I'll swap and try again.

**Comment by davthomaspilot** [comment]
Got the I2C straightened out.  

Deduced the t the GPIO assignment for the flow switch by turning a pump on/off and observing what gpio changed state using "gpio readall".

Still thinking about how I figure which GPIO control the dosing pumps...

**Comment by davthomaspilot** [comment]
Found the documentation for the relay board, and that told me the pump relays were on GPIO20 and GPIO21.  (don't know which is the chlorine versus the acid yet).

I can add pins for the relays using the GPIO - PINOUTs tab, but I'm confused about "triggers".  Do I need to add one of those for each of the pump relay pins?  If so, what event name do I choose from the pull-down list?  Does it need to match an event name that will be sent by njsPC? 

Also, I the Chemistry-Hardware menu doesn't keep the "REM Feed" boxes checked.   The checkbox under the PH probe clears when I do the save while the one under the ORP probe remains checked.  However, they are both cleared when I leave the page and revisit it.

Yet, the ph and ORP readings seem reasonable in the Chemistry Panel on the main screen of the dashPanel.

**Comment by davthomaspilot** [comment]
Actually, the ph and ORP readings aren't updating in the dashPanel.  But I do see them updating on the REM web page interface.

I suspect it's related to the REM Feed checkboxes not staying enabled.

Help!

**Comment by davthomaspilot** [comment]
Ok, figured that one out too.  

The body changes when you leave the tab and revisit it.   

In my case, I have three bodies configured--Pool, Spa, and Pool/Spa.  

I don't think that makes sense -- I didn't know what I was doing when I configure it.  And, I didn't realize the  "REM Feed" checkbox can be different for each body.

So, I was actually checking the REM Feed boxes on a different body than what comes up by default when you visit the page.

I'll have to go back and refresh my memory on what multiple bodies imply for njsPC and REM.

**Reply by tagyoureit** [reply]
It doesn't work that way.  The pool/spa/(pool/spa) drop down determines which body of water the REM Chem is assigned to.  If you have a single REM chem and a shared pool/spa then you should have it set to pool/spa.  If you have a dual body pool with separate spa then you would have two REM chems installed and one woulde be assigned to the pool body and the other to the spa body.  

If you set it for a body of water, but it doesn't save upon reload, that could be a bug.  But for the most part you shouldn't change this once it's set and you shouldn't try to set different values for different bodies of water because you'll just be overwriting the previous settings.
